app/src/pages/market_home_page.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import { useWeb3Context } from 'web3-react'
+
+import { MarketHomeConnectedContainer, MarketHomeDisconnectedContainer } from '../components'
+import { ConnectedWeb3, useConnectWeb3 } from '../hooks/connectedWeb3'
+import { DisconnectedWeb3 } from '../hooks/disconnectedWeb3'
+import connectors from '../util/connectors'
+
+const MarketHomeConnectedWrapper = (props: any) => {
+  useConnectWeb3()
+
+  return (
+    <ConnectedWeb3>
+      <MarketHomeConnectedContainer {...props} />
+    </ConnectedWeb3>
+  )
+}
+
+const MarketHomeDisconnectedWrapper = (props: any) => {
+  return (
+    <DisconnectedWeb3>
+      <MarketHomeDisconnectedContainer {...props} />
+    </DisconnectedWeb3>
+  )
+}
+
+const MarketHomePage = (props: any) => {
+  const { active } = useWeb3Context()
+  const connector = localStorage.getItem('CONNECTOR')
+
+  return active || (connector && (connector as keyof typeof connectors)) ? (
```

Comment:
What does `connector as keyof typeof connectors` do? That's just meaningful at compile time; at runtime this is equivalent to `connector && connector` as far as I know. Is the cast helpful in some way?

Reply:
I added an `in` operator to validate the connector exist